src/components/dropdown-disclosure/index.tsx

```diff
@@ -0,0 +1,87 @@
+import classNames from 'classnames'
+import Disclosure, { DisclosureContent } from 'components/disclosure'
+import {
+	DropdownDisclosureActivator,
+	DropdownDisclosureButtonItem,
+	DropdownDisclosureDescriptionItem,
+	DropdownDisclosureLabelItem,
+	DropdownDisclosureLinkItem,
+	DropdownDisclosureSeparatorItem,
+} from './components'
+import { DropdownDisclosureProps } from './types'
+import { validateDropdownDisclosureChildren } from './helpers'
+import s from './dropdown-disclosure.module.css'
+
+const DropdownDisclosure = ({
+	'aria-label': ariaLabel,
+	children,
+	className,
+	color = 'primary',
+	hideChevron = false,
+	icon,
+	text,
+	listPosition = 'left',
+}: DropdownDisclosureProps) => {
+	// Check if `children` are valid
+	validateDropdownDisclosureChildren(children)
+
+	// Check for invalid props
+	const hasIcon = !!icon
+	const hasText = !!text
+	const hasLabel = !!ariaLabel
+	const isIconOnly = hasIcon && !hasText
+	const isTextOnly = !hasIcon && hasText
+
+	if (hasIcon && hasText) {
+		throw new Error(
+			'`DropdownDisclosure` does not accept both `icon` and `label`. Only provide one or the other.'
+		)
+	}
+
+	if (isIconOnly && !hasLabel) {
+		throw new Error(
+			'Icon-only `DropdownDisclosure`s require an accessible label. Provide an `aria-label` or replace the `icon` prop with the `text` prop.'
+		)
+	}
+
+	if (isTextOnly && hideChevron) {
+		throw new Error(
+			'`DropdownDisclosure`s with the `text` prop require `hideChevron` to be `false`.'
+		)
+	}
```

Comment:
These aren't just development errors. We don't want to see these conditions happen in any environment. If one happens in production and a user catches and reports it, that's good and what we want to happen. 😊

Reply:
Can you think of a scenario where these errors would make it past development/preview and into production? If these errors are thrown in production we'll render the generic error page, making the page entirely unusable. That's a significant incident for us. Additionally, a user would need to know to look in the console to see these error messages. That does not seem like an acceptable UX tradeoff to me.

These errors are something that should always be caught before production, and in the event that they somehow make it through our full test/build/deploy process we don't want to be causing an outage.

Reply:
These aren't different than any other runtime errors that can get through to production, so I don't have a specific example other than a common way production errors can pop up: code getting changed in a way that has unintended side effects that aren't noticed right away.

I don't think the solution to "the error page is too generic" is "don't throw errors in production". In these error conditions, the component is broken and in some cases it's unusable. Is there anything stopping us from improving our error UX? One idea would be to throw up a modal on top of the current page with a "report bug" button that enables users to directly notify us of exactly what happened. They wouldn't be taken to a different page, and after the modal closes they can still interact with the unaffected parts of the page.

Reply:
> These aren't different than any other runtime errors that can get through to production

They are, because we are deliberately throwing these errors.

> I don't think the solution to "the error page is too generic" is "don't throw errors in production".

Where we can, we should attempt to not deliberately throw errors in production and instead look at other alternatives, like logging to our error monitoring service. Graceful degradation should be our preferred approach.

> In these error conditions, the component is broken and in some cases it's unusable. Is there anything stopping us from improving our error UX?

Not necessarily, just priority and time 😄 It's important to consider here what we consider "broken." Our _current_ error view being displayed is a much more broken experience than a dropdown potentially being broken or unusable. I understand we can make improvements to how we handle errors that occur during rendering, but we need to consider the current state. If some subset of our pages suddenly start rendering the error view, that's an incident that will get reported to us and likely negatively impact our customers. Alternatively, if we have non-fatal logging in place, we should see that pop up in our dashboards and we can address it without major impact.

> One idea would be to throw up a modal on top of the current page with a "report bug" button that enables users to directly notify us of exactly what happened.

FWIW, we do have error monitoring, so the errors should get sent to Datadog, but it's still a poor experience today.

Reply:
@BRKalow I've added a `developmentToast` in `src/components/toast` and set up a `handleError` function in `DropdownDisclosure` that logs an error to the console and then invokes `developmentToast`. Does this work?

Reply:
This looks fantastic! Good thinking ⭐